### PR TITLE
Add SystemFonts tests

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

**Comment by hughbe** [comment]
Test Linux x64 Release Build

File: src/System.Drawing.Common/src/System/Drawing/SystemFonts.cs [hunk 1]
```diff
@@ -21,31 +19,12 @@ public static Font CaptionFont
                 var data = new NativeMethods.NONCLIENTMETRICS();
                 bool result = UnsafeNativeMethods.SystemParametersInfo(NativeMethods.SPI_GETNONCLIENTMETRICS, data.cbSize, data, 0);
 
-                if (result && data.lfCaptionFont != null)
```
Comment (stephentoub) [inline src/System.Drawing.Common/src/System/Drawing/SystemFonts.cs]: Why did you remove the != null check?  What guarantee is there that it's non-null even if result is true?

File: src/System.Drawing.Common/src/System/Drawing/SystemFonts.cs [hunk 2]
```diff
@@ -21,31 +19,12 @@ public static Font CaptionFont
...
-                    }
-                    else if (captionFont.Unit != GraphicsUnit.Point)
-                    {
-                        captionFont = FontInPoints(captionFont);
-                    }
+                    captionFont = GetFontFromData(data.lfCaptionFont);
                 }
 
-                captionFont.SetSystemFontName("CaptionFont");
+                captionFont.SetSystemFontName(nameof(CaptionFont));
```
Comment (stephentoub) [inline src/System.Drawing.Common/src/System/Drawing/SystemFonts.cs]: Isn't this going to null-ref if result is false?

File: src/System.Drawing.Common/src/System/Drawing/SystemFonts.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Drawing.Common/src/System/Drawing/SystemFonts.cs]: I see, you added a null check to GetFontFromData... ok.

File: src/System.Drawing.Common/src/System/Drawing/SystemFonts.cs [hunk 3]
```diff
@@ -377,43 +245,66 @@ private static Font FontInPoints(Font font)
...
+            if (font == null)
+            {
+                return DefaultFont;
+            }
+            else if (font.Unit != GraphicsUnit.Point)
+            {
+                return FontInPoints(font);
+            }
+
+            return font;
```
Comment (stephentoub) [inline src/System.Drawing.Common/src/System/Drawing/SystemFonts.cs]: These last lines could be consolidated to:
```C#
return
    font != null ? DefaultFont :
    font.Unit != GraphicsUnit.Point : FontInPoints(font) :
    font;
```

File: src/System.Drawing.Common/src/System/Drawing/SystemFonts.cs (hunk 2 above)
Comment (hughbe) [inline src/System.Drawing.Common/src/System/Drawing/SystemFonts.cs]: Yes - would this have happened anyway in the old implementation?

File: src/System.Drawing.Common/tests/SystemFontsTests.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            yield return Font(() => SystemFonts.StatusFont, "StatusFont", "Segoe UI");
+
+            bool isArabic = (GetSystemDefaultLCID() & 0x3ff) == 0x0001;
+            yield return Font(() => SystemFonts.DefaultFont, "DefaultFont", isArabic ? "Tahoma" : "Microsoft Sans Serif");
+
+            bool isJapanese = (GetSystemDefaultLCID() & 0x3ff) == 0x0011;
+            yield return Font(() => SystemFonts.DialogFont, "DialogFont", isJapanese ? "Microsoft Sans Serif" : "Tahoma");
+        }
+
+        public static object[] Font(Func<Font> fontThunk, string systemFontName, string windowsFontName) => new object[] { fontThunk, systemFontName, windowsFontName };
```
Comment (stephentoub) [inline src/System.Drawing.Common/tests/SystemFontsTests.cs]: Nit: Rather than `fontThunk`, I'd suggest `createFont` or `getFont` or something like that.

File: src/System.Drawing.Common/src/System/Drawing/SystemFonts.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Drawing.Common/src/System/Drawing/SystemFonts.cs]: Yes

**Comment by hughbe** [comment]
Thanks for reviewing - I addressed your feedback

**Comment by mellinoe** [comment]
Thanks for the continued contributions, @hughbe . Much appreciated.
